lib/testing/testDataCheck.hpp

```diff
@@ -126,30 +121,30 @@ void testDataCheck<A_Type>::main_thread() {
                     error = true;
                     num_errors += 1;
                     if (num_errors < max_num_errors) {
-                        FATAL_ERROR("{:s}[{:d}][{:d}] != {:s}[{:d}][{:d}]; values: ({:f}, {:f}), "
-                                    "epsilon: {:f}, "
-                                    "abs(x-y): {:f}, epsilon * abs(x+y): {:f}",
-                                    first_buf->buffer_name, first_buf_id, i,
-                                    second_buf->buffer_name, second_buf_id, i, v1, v2, epsilon,
-                                    std::abs((float)(first_value - second_value)),
-                                    epsilon * std::abs((float)(first_value + second_value)));
+                        ERROR("{:s}[{:d}][{:d}] != {:s}[{:d}][{:d}]; values: ({:f}, {:f}), "
+                              "epsilon: {:f}, "
+                              "abs(x-y): {:f}, epsilon * abs(x+y): {:f}",
+                              first_buf->buffer_name, first_buf_id, i,
+                              second_buf->buffer_name, second_buf_id, i, v1, v2, epsilon,
+                              std::abs((float)(first_value - second_value)),
+                              epsilon * std::abs((float)(first_value + second_value)));
                     }
                 }
-            } else { // N2 numbers are int
-                // INFO("Checking non float numbers-----------");
+            } else {
                 if (first_value != second_value) {
-                    if (num_errors++ < max_num_errors)
+                    error = true;
```

Comment:
This `error` flag is kind of redundant, since `num_errors > 0` is equivalent, but I'll leave it to you if you want to remove before merging, or just leave it.